packages/edit-site/src/index.js

```diff
@@ -38,7 +40,21 @@ export async function reinitializeEditor( target, settings ) {
 	// define what's being edited. When visiting via the dashboard link, these
 	// won't be present. Do a client side redirect to the 'homepage' if that's
 	// the case.
-	await redirectToHomepage( settings.siteUrl );
+	try {
+		await redirectToHomepage( settings.siteUrl );
+	} catch ( error ) {
+		render(
+			<ErrorBoundaryWarning
+				message={ __(
+					'The editor is unable to find a block template for the homepage.'
+				) }
+				error={ error }
+				dashboardLink="index.php"
```

Comment:
I'm not sure if this link is guaranteed to be the dashboard link or should we use something `adminUrl` from server side?

Reply:
I'm not sure either, but there's existing released code that does this in the Navigation Sidebar:
https://github.com/WordPress/gutenberg/blob/bfddbdcd2f8f9b06870de5a38171928f337acfc4/packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js#L80-L84

So I think it must be ok.